bundle/src/main/java/com/adobe/acs/commons/wcm/impl/AemEnvironmentIndicatorFilter.java

```diff
@@ -193,10 +219,9 @@ public final void doFilter(final ServletRequest servletRequest, final ServletRes
                             ? capturedResponse.getBufferedServletOutput().getBufferedString()
                             : null;
 
-            
-                            
             if (contents != null
-                    && StringUtils.contains(response.getContentType(), "html")) {
+                    && StringUtils.contains(response.getContentType(), "html")
+                    && innerFilterAcceptsInjection(request)) {
```

Comment:
The `accepts`-method (which is checked before this try/finally block, for an early-exit) still obtains the `WCMMode`, which will never yield a value other than `DISABLED`. This was probably again masked by the unit tests due to the invalid use of spying/mocking internals of this class.

Reply:
I rewrote the unittests for the accept method.